staging/src/k8s.io/apiserver/pkg/server/options/helpers.go

```
@@ -0,0 +1,107 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package options
+
+import (
+	"errors"
+	"io/ioutil"
+
+	yaml "github.com/ghodss/yaml"
+	"github.com/golang/glog"
+
+	"k8s.io/apiserver/pkg/storage/value"
+	aestransformer "k8s.io/apiserver/pkg/storage/value/encrypt/aes"
+)
+
+// Used for parsing command line parameters for selecting transformer
+type EncryptionProviderConfig struct {
+	Transf *value.Transformer
+	name   string
+}
+
+func (e EncryptionProviderConfig) Set(filepath string) error {
+	data, err := ioutil.ReadFile(filepath)
+	if err != nil {
+		return err
+	}
+
+	var providers []map[string]interface{}
+	yaml.Unmarshal(data, &providers)
+
+	// The final transformers which will be wrapped inside a location transformer, and then a mutable transformer
+	transformers := []value.LocationTransformer{}
+
+	// For each provider listed in config file
+	for _, provider := range providers {
+
+		// Parse the config map to get configuration
+		providerConfig, err := parseProviderInfo(provider)
+		if err != nil {
+			glog.Warningf(err.Error())
```

Why not return an error here? It seems like the config file is unparseable, but just a warning is logged? Wouldn't returning an error cause the process to not start?